saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/DefaultSaml2AuthenticatedPrincipal.java

```diff
@@ -26,12 +26,12 @@
  * @author Clement Stoquart
  * @since 5.2.2
  */
-class SimpleSaml2AuthenticatedPrincipal implements Saml2AuthenticatedPrincipal, Serializable {
+public class DefaultSaml2AuthenticatedPrincipal implements Saml2AuthenticatedPrincipal, Serializable {
 
 	private final String name;
 	private final Map<String, List<Object>> attributes;
 
-	SimpleSaml2AuthenticatedPrincipal(String name, Map<String, List<Object>> attributes) {
+	DefaultSaml2AuthenticatedPrincipal(String name, Map<String, List<Object>> attributes) {
```

Comment:
I'm thinking the constructor should be public as well so that applications can use it in their own authentication providers.

Reply:
Ah, missed that.